src/metabase/lib/drill_thru/object_details.cljc

```diff
@@ -55,10 +15,11 @@
 
   Contrast [[metabase.lib.drill-thru.fk-filter/fk-filter-drill]], which filters this query to only those rows with a
   specific value for a FK column."
-  [query                              :- ::lib.schema/query
-   stage-number                       :- :int
-   {:keys [column value] :as context} :- ::lib.schema.drill-thru/context]
-  (when (and column
-             (some? value))
-    (object-detail-drill-for query stage-number context
-                             (> (count (lib.metadata.calculation/primary-keys query)) 1))))
+  [query        :- ::lib.schema/query
+   stage-number :- :int
+   context      :- ::lib.schema.drill-thru/context]
+  (some (fn [f]
+          (f query stage-number context))
+        [lib.drill-thru.fk-details/fk-details-drill
+         lib.drill-thru.pk/pk-drill
+         lib.drill-thru.zoom/zoom-drill]))
```

Comment:
These are in a different order than the conditions were checked previously - I assume that's on purpose? A comment about the order would be a good idea here.

Reply:
yeah I did initially change the order because it mattered but after reworking a few things it's not important anymore